packages/codebytes/src/drawers.tsx

```diff
@@ -0,0 +1,128 @@
+import { FlexBox, IconButton } from '@codecademy/gamut';
+import {
+  ArrowChevronLeftIcon,
+  ArrowChevronRightIcon,
+} from '@codecademy/gamut-icons';
+import styled from '@emotion/styled';
+import React, { useState } from 'react';
+
+const DrawerLabel = styled.span`
+  padding: 0.875rem 0.5rem;
+`;
+
+const LeftDrawerIcon = styled(ArrowChevronLeftIcon)<{ open?: boolean }>`
+  transition: transform 0.2s ease-in-out;
+`;
+const RightDrawerIcon = LeftDrawerIcon.withComponent(ArrowChevronRightIcon);
+
+const Drawer = styled(FlexBox)<{ open?: boolean; hideOnClose?: boolean }>`
+  position: relative;
+  ${({ open, hideOnClose }) => `
+    flex-basis: ${open ? '100%' : '0%'};
+    ${!open && hideOnClose ? 'visibility: hidden;' : ''}
+    transition: flex-basis 0.2s ${
+      open ? 'ease-out' : 'ease-in, visibility 0s 0.2s'
+    };
+
+    ${LeftDrawerIcon}, ${RightDrawerIcon} {
+      transform: rotateZ(${open ? '0' : '180'}deg)};
+    }
+  `}
+`;
+
+export type DrawersProps = {
+  leftChild: React.ReactNode;
+  rightChild: React.ReactNode;
+};
+
+export const Drawers: React.FC<DrawersProps> = ({ leftChild, rightChild }) => {
+  const [show, setShow] = useState<'left' | 'right' | 'both'>('both');
+
+  let buttonLabelL = 'hide code';
+  let buttonLabelR = 'hide output';
+
+  if (show === 'left') {
+    buttonLabelL = buttonLabelR = 'show output';
+  } else if (show === 'right') {
+    buttonLabelL = buttonLabelR = 'show code';
+  }
+
+  return (
+    <>
+      <FlexBox>
+        <Drawer
+          open={show !== 'right'}
+          alignItems="center"
+          flexWrap="nowrap"
+          textAlign="left"
+          borderRight="1"
+          borderColor="gray-900"
+          px="8"
```

Comment:
nit
```suggestion
          px={8}
```

Reply:
weirdly, i am getting linter errors if it is not a string, i can look more into why

Reply:
weird. not a big deal in this case, but i'd be slightly concerned that then we're not getting the responsive capability and proper type checks from gamut

Reply:
spoke to josh and cass about this today, and we resolved that there's some additional setup in gamut and the monolith that allows us to pass in numerical values. The theme.d.ts file is needed to extend the current gamut theme!

Reply:
gotcha, are you going to add that in this pr? ticket it out? or josh is taking care of it?

Reply:
oh wait it looks like you already made the change?

Reply:
yes, it's all added in this pr!